worker/rhpv2.go

```diff
@@ -155,6 +160,65 @@ type Session struct {
 	mu          sync.Mutex
 }
 
+// transport wraps rhpv2.Transport to make sure we always apply the context's
+// deadline as read/write deadline to the transport's underlying connection.
+type transport struct {
+	t *rhpv2.Transport
+}
+
+func (t *transport) writeRequest(ctx context.Context, rpcID types.Specifier, req rhpv2.ProtocolObject) (err error) {
+	if deadline, ok := ctx.Deadline(); ok {
```

Comment:
Hm yeah. I added a very pessimistic timeout in `withHosts` to ensure unlocking/force closing sessions don't deadlock (which is the whole reason I made this PR by the way)

Reply:
I considered wrapping `jape.Mux` with a timeout handler but we can't do that and I want to avoid applying a timeout in every location where it might make sense, so I think `withHosts` is a good place to put it.